guides/integrate-nextauth.md

```diff
@@ -0,0 +1,327 @@
+          // -- 4
+          window.location.href = redirectURL;
+        }).catch(error => {
+          setBiAuthenticateResult(error.toString());
+        });
+      }
+    }
+    authenticate().catch(console.error);
+  }, []);
+
+  // -- 3
+  async function biAuthenticate(url: string): Promise<string> {
+    const BeyondIdentityEmbeddedSdk = await import("@beyondidentity/bi-sdk-js");
+    let embedded = await BeyondIdentityEmbeddedSdk.Embedded.initialize();
+
+    // Display passkeys so user can select one
+    let passkeys = await embedded.getPasskeys();
+    let promptText = passkeys.map((passkey, index) => {
+      return `${index}: ${passkey.identity.username}`;
+    }).join("\n");
+    let selectedIndex = parseInt(prompt(promptText, "index")!!);
+    if (selectedIndex >= 0 && selectedIndex < passkeys.length) {
+      let selectedId = passkeys[selectedIndex].id;
+      // Perform authentication using selected id
+      let result = await embedded.authenticate(url, selectedId);
+      return Promise.resolve(result.redirectURL);
+    } else {
+      // This will fail in core as it won't match to any id
+      return Promise.resolve("unknown_id");
+    }
+  }
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100vh',
+      }}
+    >
+      <div className="container">
+        <div className="row">
+          <div className="d-flex justify-content-center">
+            <div className="spinner-border" role="status">
+              <span className="sr-only"></span>
+            </div>
+          </div>
+        </div>
+        <div className="row">
+          {
+            biAuthenticateResult.length > 0 &&
+            <div className="row row-cols-1 row-cols-md-1 mt-3">
+              <div className="col">
+                <code>
+                  {JSON.stringify(biAuthenticateResult, null, 2)}
+                </code>
+              </div>
+            </div>
+          }
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default BIAuthenticate;
+```
+
+What's happening here?
+
+1. The `useEffect` is only called once on page load. In this function, we initialize the Beyond Identity SDK and use `embedded.isAuthenticateUrl` to check if the current page that was redirected to is in fact a valid `bi-authenticate` URL.
+2. If the URL is valid, we pull the URL using `window.location.href` and pass that directly into `biAuthenticate` in step 3.
+3. `biAuthenticate` calls `embedded.authenticate` with a valid `bi-authenticate` URL. This function performs a challenge/response against a passkey bound to your browser. Note that the callback in `embedded.authenticate` contains logic in order to prompt a user to select a passkey if there is more than one.
+4. Finally, the response of `embedded.authenticate` contains a `redirectURL`. Follow this redirectURL to complete the OAuth/OIDC flow.
+
+### Token Exchange
+
```

Comment:
It looks like old pngs in `static/assets` were left from the previous iteration too.

Reply:
Do you mean [static/assets/nextauth-application-config.png](https://github.com/gobeyondidentity/developer-docs/pull/199/files#diff-62e0d33b52eb1e697619700ec5fcef44b16782107c91af6e64d96a460462e9d9) and [static/assets/nextauth-authenticator-config.png](https://github.com/gobeyondidentity/developer-docs/pull/199/files#diff-91251ca5b3ac30aeca567b644235f7d88e6eedf904a4b1675469a70f0edab20a)? Yes, I suppose I am not using these anymore.

Reply:
Yep those

Reply:
Done.